### PR TITLE
MAINT: Removed conditionals that are always false in datetime_strings.c

### PR DESCRIPTION
**Pull request by kawochen**
These conditionals always evaluate to `0`.

### CODE REVIEW

**Comment by jaimefrio** [comment]
Can you ellaborate on why negative values are not possible?

File: numpy/core/src/multiarray/datetime_strings.c [hunk 1]
```diff
@@ -484,7 +484,7 @@ parse_iso_8601_datetime(char *str, Py_ssize_t len,
     if (sublen >= 2 && isdigit(substr[0]) && isdigit(substr[1])) {
         out->hour = 10 * (substr[0] - '0') + (substr[1] - '0');
```
Comment (kawochen) [inline numpy/core/src/multiarray/datetime_strings.c]: `substr[0]` and `substr[1]` are `char` digits (`isdigit`), of which `'0'` is the smallest, so `out->hour` can only be as small as `10 * 0 + 0 = 0`

**Comment by charris** [comment]
LGTM. One of the fixes looks like is also fixes a cut and paste error.

**Comment by charris** [comment]
Thanks @kawochen .

**Pull request merged**